PWGGA/GammaConv/AliAnalysisTaskGammaConvV1.cxx

```diff
@@ -2563,280 +2565,106 @@ void AliAnalysisTaskGammaConvV1::UserExec(Option_t *)
-        AliFatal(Form("ERROR: LoadElecDeDxPostCalibration returned kFALSE for %d despite being requested!",fInputEvent->GetRunNumber()));
+
+    Double_t lPt = thePhoton->Pt();
+    Float_t lWeightMatBudgetGamma = (fDoMaterialBudgetWeightingOfGammasForTrueMesons &&
+                                     fiPhotonCut->GetMaterialBudgetWeightsInitialized())
+        ? fiPhotonCut->GetMaterialBudgetCorrectingWeightForTrueGamma(thePhoton, fInputEvent->GetMagneticField())
+        : 1.;
+
+    Double_t lTotalWeight = (fDoCentralityFlat > 0) ? fWeightJetJetMC*lWeightMatBudgetGamma*fWeightCentrality[fiCut]
+                                                    : fWeightJetJetMC*lWeightMatBudgetGamma;
+    fHistoConvGammaPt[fiCut]->Fill(lPt, lTotalWeight);
+
+    if (fDoPhotonQA > 0 && fIsMC < 2){
+      fHistoConvGammaPsiPairPt[fiCut]->Fill(thePhoton->GetPsiPair(),lPt, lTotalWeight);
+      fHistoConvGammaR        [fiCut]->Fill(thePhoton->GetConversionRadius(), lTotalWeight);
+      fHistoConvGammaEta      [fiCut]->Fill(thePhoton->Eta(), lTotalWeight);
+      fHistoConvGammaPhi      [fiCut]->Fill(thePhoton->Phi(), lTotalWeight);
+
+      if (fDoPhotonQA == 4 || fDoPhotonQA == 5){
+        fHistoConvGammaInvMass    [fiCut]->Fill(thePhoton->GetMass(), lTotalWeight);
+        fHistoConvGammaInvMassReco[fiCut]->Fill(GetOriginalInvMass(thePhoton, fInputEvent), lTotalWeight);
       }
     }
 
-
-    if( fIsMC > 0 && signalRejection){
-
-      Int_t isPosFromMBHeader = fiEventCut->IsParticleFromBGEvent(PhotonCandidate->GetMCLabelPositive(), fMCEvent, fInputEvent);
-
-      if (signalRejection==3){
-        Int_t isNegFromMBHeader = fiEventCut->IsParticleFromBGEvent(PhotonCandidate->GetMCLabelNegative(), fMCEvent, fInputEvent);
-        isFromSelectedHeader = (isNegFromMBHeader+isPosFromMBHeader)==4;
+    if (fDoPhotonQA == 2 || fDoPhotonQA == 5){
+      if ((fIsHeavyIon == 1                             && lPt > 0.399 && lPt < 12.) ||
+          (fiPhotonCut->GetSingleElectronPtCut() < 0.04 && lPt > 0.099 && lPt < 16.) ||
+          (                                                lPt > 0.299 && lPt < 16.))
+      {
+        fPtGamma     = lPt;
+        fDCAzPhoton  = thePhoton->GetDCAzToPrimVtx();
+        fRConvPhoton = thePhoton->GetConversionRadius();
+        fEtaPhoton   = thePhoton->GetPhotonEta();
+        iCatPhoton   = thePhoton->GetPhotonQuality();
+        tESDConvGammaPtDcazCat[fiCut]->Fill();
       }
-      else{ // 1,2,4
-        if (!isPosFromMBHeader) continue;
-        Int_t isNegFromMBHeader = fiEventCut->IsParticleFromBGEvent(PhotonCandidate->GetMCLabelNegative(), fMCEvent, fInputEvent);
-        if (!isNegFromMBHeader) continue;
+    }
+  }; // end of lambda fillHistosAndTree()
 
-        if (signalRejection!=2) isFromSelectedHeader = (isNegFromMBHeader+isPosFromMBHeader)==4;
-      }
+  // ProcessPhotonCandidates() starts here
+  if(fiPhotonCut->GetDoElecDeDxPostCalibration()){
```

Comment:
The Photon QA should have the post calibration?

Reply:
> The Photon QA should have the post calibration?
mm sorry, I dont understand the question. This if clause has been taken from the existing code. Don't know if this answers your question?